src/module.c

```diff
@@ -7363,13 +7363,20 @@ RedisModuleBlockedClient *RM_BlockClientOnKeys(RedisModuleCtx *ctx, RedisModuleC
 
 /* This function is used in order to potentially unblock a client blocked
  * on keys with RedisModule_BlockClientOnKeys(). When this function is called,
- * all the clients blocked for this key will get their reply_callback called.
- *
- * Note: The function has no effect if the signaled key doesn't exist. */
+ * all the clients blocked for this key will get their reply_callback called. */
 void RM_SignalKeyAsReady(RedisModuleCtx *ctx, RedisModuleString *key) {
     signalKeyAsReady(ctx->client->db, key, OBJ_MODULE);
 }
 
+/* Same as RedisModule_SignalKeyAsReady, but takes a db-id instead of
+ * a RedisModuleCtx */
+int RM_SignalKeyAsReadyByDbId(int dbid, RedisModuleString *key) {
```

Comment:
i'd like to mention that it was just by chance that all that `signalKeyAsReady` needs is a db object, and that `RedisModuleKeyOptCtx` has an API for retrieving the db-id...

perhaps a more general approach would be to have something like `RM_CreateContextFromKeyOptCtx` (so why did we bother to invent `RedisModuleKeyOptCtx`..?)